okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -432,125 +422,111 @@ void removePendingStream(OkHttpClientStream pendingStream) {
 
   @Override
   public Runnable start(Listener listener) {
-    this.listener = Preconditions.checkNotNull(listener, "listener");
-
-    if (enableKeepAlive) {
-      scheduler = SharedResourceHolder.get(TIMER_SERVICE);
-      keepAliveManager = new KeepAliveManager(
-          new ClientKeepAlivePinger(this), scheduler, keepAliveTimeNanos, keepAliveTimeoutNanos,
-          keepAliveWithoutCalls);
-      keepAliveManager.onTransportStarted();
-    }
-
-    frameWriter = new AsyncFrameWriter(this, serializingExecutor);
-    outboundFlow = new OutboundFlowController(this, frameWriter, initialWindowSize);
-    // Connecting in the serializingExecutor, so that some stream operations like synStream
-    // will be executed after connected.
-    serializingExecutor.execute(new Runnable() {
-      @Override
-      public void run() {
-        if (isForTest()) {
-          if (connectingCallback != null) {
-            connectingCallback.run();
-          }
-          clientFrameHandler = new ClientFrameHandler(testFrameReader);
-          executor.execute(clientFrameHandler);
-          synchronized (lock) {
-            maxConcurrentStreams = Integer.MAX_VALUE;
-            startPendingStreams();
-          }
-          frameWriter.becomeConnected(testFrameWriter, socket);
-          connectedFuture.set(null);
-          return;
-        }
+    synchronized (lock) {
```

Comment:
This lock does I/O while held. That would cause `shutdownNow()` to block.

I'm thinking we can take a page from the previous code: let's go ahead and create `AsyncSink` and delay creating `Okio.sink(sock)`. We can schedule work on the `serializingExecutor` to create the connection and then afterward use a simple setter to set it on the `AsyncSink`.